steps/17_forecasting_horizon_freq/step.md

```diff
@@ -149,4 +149,33 @@ Preferably:
 * time-like `pandas.Index` objects
 * `freq` strings as used in `pandas`
 
-The function `get_freq` can then be invoked at the start of any method with a `freq` input to convert different possible input options to a uniform `freq` representation.
\ No newline at end of file
+The function `get_freq` can then be invoked at the start of any method with a `freq` input to convert different possible input options to a uniform `freq` representation.
+
+### 3. turn `cutoff` into a `pd.Index`
+
+Time-like `pd.Index` objects carry a `freq` attribute and will also not lose it in the above-mentioned deprecation.
+
+This motivates the following solution: turning forecasters' `cutoff` into a one-element `pd.Index` with `freq`.
+
+The solution combines the following:
+
+* the possibility to pass `freq` strings to methods of `ForecastingHorizon`, like in 2.
+* turning `_cutoff` and possibly `cutoff` attributes into  length 1 `pandas.Index` objects
+* extend utilities that take a cutoff so they can input a length 1 index with potential `freq` attribute
+* extend utilities sthat return a cutoff so they can return a length 1 index that stores `freq`
+* in private interfaces, use `pd.Index` typed `_cutoff` instead of `cutoff`
+
+The advantage of this solution is that existing logic and arguments can be kept.
+
+If utilities are extended with polymorphic signatures, downwards compatibility is also maintained. 
+This means:
+* where utilities accept index element input, they should also accept length 1 `pd.Index` input
+* where utilities return index element output, they should be able to return lenght 1 `pd.Index` input
+* which of the two outputs is returned is controlled by a `return_index` boolean argument (default=`False`)
+
+Optionally, `cutoff` in index element format can be deprecated in favour of `_cutoff`.
+
+Utilities affected by this change:
```

Comment:
I do not see any discussion about the effect on public methods like `ForecastingHorizon.to_absolute` or `.to_relative`. These methods consume `cutoff` which is provided by a user, that is we have no control over it. We can do whatever acrobatics internally to avoid changing public interfaces, but it does not solve the problem of a user passing `pd.Timestamp` as a `cutoff` to `.to_relative`.

Reply:
`ForecastingHorizon` would be as in part 2, as the first bullet point says, i.e., as in your proposed solution.

In the case where the user passes `pd.Timestamp`, in the future we will also expect them to pass a `freq` string at some point before it is required, e.g., anytime in or before `to_absolute` is being called, like in your write-up.